orttraining/orttraining/eager/ort_aten.cpp

```diff
@@ -1204,6 +1247,257 @@ at::Tensor squeeze(
+    ort_input_1_self = CastToType(invoker, ort_input_1_self, *promoted_type);
+  }
+
+  // resize the output and then create output ort value to be updated.
+  onnxruntime::TensorShape out_shape;
+  resize_output(
+    invoker,
+    dynamic_cast<ORTTensorImpl*>(out.unsafeGetTensorImpl()),
+    BroadcastShape("add_out", ort_input_1_self, ort_input_0_other, out_shape));
+
+  auto ort_input_out = create_ort_value(invoker, out);
+
+  std::vector<OrtValue> ort_outputs_1_Add(1);
+  if (*promoted_type == out.scalar_type()) {
+    ort_outputs_1_Add[0] = ort_input_out;
+  }
+
+  status = invoker.Invoke("Add", {
+    std::move(ort_input_1_self),
+    std::move(ort_outputs_0_Mul[0]),
+  }, ort_outputs_1_Add, nullptr);
+  CHECK_STATUS(status);
+
+  if (*promoted_type != out.scalar_type()) {
+    CastToType_out(invoker, ort_outputs_1_Add[0], ort_input_out, out.scalar_type());
+  }
+  return out;
+}
+
+// aten::sub.out(Tensor self, Tensor other, *, Scalar alpha=1, Tensor(a!) out) -> Tensor(a!)
+at::Tensor& sub_out(
+  const at::Tensor& self,
+  const at::Tensor& other,
+  // *,
+  const at::Scalar& alpha,
+  at::Tensor& out) {
+  ORT_LOG_FN(self, other, alpha, out);
+
+  auto promoted_type = PromoteScalarTypesWithCategory({other.scalar_type(),self.scalar_type()}, {alpha.type()});
+
+  if (
+    !IsSupportedType(alpha, {at::kBFloat16,at::kByte,at::kDouble,at::kFloat,at::kHalf,at::kInt,at::kLong,at::kShort}) ||
+    !IsSupportedType(other, {at::kBFloat16,at::kByte,at::kDouble,at::kFloat,at::kHalf,at::kInt,at::kLong,at::kShort}) ||
+    !IsSupportedType(self, {at::kBFloat16,at::kByte,at::kDouble,at::kFloat,at::kHalf,at::kInt,at::kLong,at::kShort}) ||
+    !c10::canCast(*promoted_type, out.scalar_type())) {
+    return at::native::call_fallback_fn<
+       &at::native::cpu_fallback,
+      ATEN_OP(sub_out)>::call(self, other, alpha, out);
+  }
+  auto& invoker = GetORTInvoker(self.device());
+
+  auto ort_input_0_alpha = create_ort_value(invoker, alpha);
+  if (alpha.type() != *promoted_type){
+    ort_input_0_alpha = CastToType(invoker, ort_input_0_alpha, *promoted_type);
+  }
+  auto ort_input_0_other = create_ort_value(invoker, other);
+  if (other.scalar_type() != *promoted_type){
+    ort_input_0_other = CastToType(invoker, ort_input_0_other, *promoted_type);
+  }
+
+  std::vector<OrtValue> ort_outputs_0_Mul(1);
+
+  auto status = invoker.Invoke("Mul", {
+    std::move(ort_input_0_alpha),
+    std::move(ort_input_0_other),
+  }, ort_outputs_0_Mul, nullptr);
+  CHECK_STATUS(status);
+
+  auto ort_input_1_self = create_ort_value(invoker, self);
+  if (self.scalar_type() != *promoted_type){
+    ort_input_1_self = CastToType(invoker, ort_input_1_self, *promoted_type);
+  }
+
+  // resize the output and then create output ort value to be updated.
+  onnxruntime::TensorShape out_shape;
+  resize_output(
+    invoker,
+    dynamic_cast<ORTTensorImpl*>(out.unsafeGetTensorImpl()),
+    BroadcastShape("add_out", ort_input_1_self, ort_input_0_other, out_shape));
```

Comment:
this would be sub_out if you're keeping the param.  Consider __func__ instead of hardcoding.

Reply:
Thank you!